integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py

```diff
@@ -75,11 +90,17 @@ def _generate_embeddings(self, documents: List[Document], batch_size: int) -> Li
         """
         Batches the documents and generates the embeddings.
         """
-        batches = [documents[i : i + batch_size] for i in range(0, len(documents), batch_size)]
+        if self._progress_bar and tqdm_imported:
+            batches = [documents[i : i + batch_size] for i in range(0, len(documents), batch_size)]
+            progress_bar = tqdm
+        else:
+            # no progress bar
+            progress_bar = _alt_progress_bar  # type: ignore
+            batches = [documents]
```

Comment:
Is this correct? It seems strange to me that batching is ignored if `tqdm` is not shown. 🤔

Reply:
Yes - our server side batch size is currently 256. 
We run a internal mapping / mirco-bacthing loop of size 256, but have no progress bar. Therefore, there is no upper limit, you could also set batch_size to 100_000

Reply:
Ok, good for me if it's good for you. 👍